src/scripts/main.js

```diff
@@ -1,3 +1,381 @@
+    showNotification('Salary is required', 'error');
+
+    return false;
+  }
+
+  return true;
+};
+
+const addComa = (value) => {
+  const improvedValue = ('' + value).split('').reverse();
+  let result = '';
+
+  for (let i = 0; i < improvedValue.length; i++) {
+    if (improvedValue.length >= 3) {
+      if (i % 3 === 0) {
+        result += ',';
+      }
+      result += improvedValue[i];
+    } else {
+      return value;
+    }
+  }
+
+  result = result.split('').reverse().join('');
+
+  if (result.endsWith(',')) {
+    result = result.slice(0, -1);
+  }
+
+  return result;
+};
+
+button.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  const formData = new FormData(form);
+  const formDataObj = Object.fromEntries(formData);
+
+  if (validateInput(formDataObj)) {
+    const tr = document.createElement('tr');
+
+    for (const key in formDataObj) {
+      const td = document.createElement('td');
+
+      td.setAttribute('data-qa', key);
+      td.setAttribute('data-row', key);
+
+      const value = formDataObj[key];
+
+      if (key === 'salary') {
+        td.textContent = `$${addComa(value)}`;
+      } else {
+        td.textContent = value;
+      }
+      tr.appendChild(td);
+    }
+    tbody.appendChild(tr);
+    showNotification('You added the new employee', 'success');
+    editRow(tr);
+
+    for (const input of form) {
+      input.value = '';
+    }
+  }
+  updateTableRows();
+});
+
+const updateTableRows = () => {
+  tableRows = document.querySelectorAll('tbody tr');
+
+  tableRows.forEach((row, i) => {
+    row.addEventListener('click', () => {
+      const selectedRow = tableRows.find((r) => r.classList.contains('active'));
+
+      if (indexRow !== i) {
+        if (selectedRow) {
+          selectedRow.classList.remove('active');
+        }
+        row.classList.add('active');
+        indexRow = i;
+      }
+    });
+  });
+};
+
+updateTableRows();
+
+const validateField = (field, value) => {
+  switch (field) {
+    case 'name':
+      if (value.length < 4 || !value.trim()) {
+        showNotification('Name must be at least 4 letters', 'error');
+
+        return false;
+      }
+      break;
+
+    case 'position':
+      if (value.length === 0 || !value.trim()) {
+        showNotification('Position is required', 'error');
+
+        return false;
+      }
+      break;
+
+    case 'office':
+      if (value.length === 0 || !value.trim()) {
+        showNotification('Country is required', 'error');
+
+        return false;
+      }
+      break;
+
+    case 'age':
+      if (
+        value < 18 ||
+        value > 90 ||
+        !value.trim() ||
+        value.toLowerCase() !== value.toUpperCase()
+      ) {
+        showNotification('Age must be between 18 and 90', 'error');
+
+        return false;
+      }
+      break;
+    case 'salary':
+      if (
+        value.length === 0 ||
+        !value.trim() ||
+        +value < 0 ||
+        value.toLowerCase() !== value.toUpperCase()
```

Comment:
The condition `value.toLowerCase() !== value.toUpperCase()` is not necessary for validating numeric fields like 'age' and 'salary'. This condition is typically used to check if a string contains alphabetic characters, which is not applicable here.